common/common.h

```diff
@@ -276,13 +300,93 @@ struct gpt_params {
     bool batched_bench_output_jsonl = false;
 };
 
-void gpt_params_parse_from_env(gpt_params & params);
-void gpt_params_handle_model_default(gpt_params & params);
+struct llama_arg {
```

Comment:
Should move the method implementations in the .cpp, to avoid building the same code in all examples.

Can also move all `llama_arg` related stuff in `common/arg.h,.cpp`. Can be in a follow-up PR

Reply:
Currently doing this will break `gen-docs`, since it reads the data directly from class member.

But yes I will do a follow up to split it into `common/arg.h,.cpp`